bool shelf_p0_hasgripper
bool shelf_p1_hasgripper
bool shelf_p2_hasgripper
bool shelf_p3_hasgripper
